shm/player.cpp

```diff
@@ -21,3 +22,8 @@ Cargo Player::getCargo(size_t index) {
 
     return 0;
 }
+
+size_t countFreeSpace() { 
+    size_t loadedSpace = std::accumulate(ship->getAllCargo.begin(), ship->getAllCargo.end(), 0);
+    return ship->getCapacity() - loadedSpace;
+}
```

Comment:
Fine solution with std::accumulate :) Please change "ship" to "ship_". Should we check if ship is nullptr?

Reply:
good point